conduct.md

```diff
@@ -24,15 +21,15 @@ As a member of the community, you agree to the following:
 
 **Encouraged:**
 
-- Be kind and courteous.
-- Respect differences of opinion and remember that every design or implementation choice carries a trade-off and numerous costs. There is seldom a single right answer.
-- Remember that everyone was new to Scala at some point. We want to encourage newcomers to join our community and learn the Scala language and ecosystem. Assume competence.
-- Show empathy towards other community members.
+- **Be kind and courteous.** We treat our fellow community members with the empathy, respect and dignity all humans deserve. Keep in mind that public communication is received by many people you don’t know, so before sending a message please ask yourself whether someone from a different context would misunderstand it.
+- **Respect differences of opinion** and remember that every design or implementation choice carries a trade-off and numerous costs. There is seldom a single right answer; we will find the best solutions by engaging in constructive discussion, with everybody bringing their unique viewpoint and experience to the table.
+- **Remember that everyone was new to Scala at some point.** We want to encourage newcomers to join our community and learn the Scala language and ecosystem. Always assume good intentions and a willingness to learn, just as you are willing to evolve your own opinion as you gain new insights.
 
 **Discouraged:**
 
+- Do not exclude people or reject their contributions based on age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, sexual identity and orientation, or other such characteristics.
```

Comment:
I disagree with this phrase because instead of saying "we are dedicated to create a safe community", it says "we shall not exclude anyone". They are very different concepts.
Diversity is never about tolerating everyone without limit. See https://en.wikipedia.org/wiki/Paradox_of_tolerance

The original phrase is sufficient.
>We are committed to providing a friendly, safe and welcoming environment for all, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, sexual identity and orientation, or other such characteristics.

Since it is the goal of the Code of Conduct and the mission of the community, it should remain on top.

Reply:
I agree with @xingyif. To me, this change doesn't seem to support the goal of making the document more friendly and welcoming. Some of the comments above said that the currently published text sound like they're scolding the reader for discrimination, but I feel like this change makes it sound _more_ like that. The original text is welcoming a diverse population into the community. This change makes it specifically about prohibiting discrimination, and only in a very narrow way (exclusion or rejection).

Reply:
@xingyif We agree on the goal of the CoC: to form the reference around which we create a welcoming community. With the proposed change, this mission statement remains front and center in the preamble, in fact it is strengthened by removing additional wording that says something else, namely the “regardless of <characteristics>” appendix.

The reason for moving that anti-discrimination list to a different part of the CoC is twofold: listing these legal-sounding terms does _not_ create a welcoming atmosphere for the reader, I posit that it has the opposite effect on most people. And as it is negative by nature, it belongs with the list of things to avoid, which is the _Discouraged_ section.

The precise wording of that sentence is up for debate. My reasoning behind the current proposal is that the unwanted effect of discrimination based on the listed characteristics is the rejection of exclusion of individuals or their contributions. If there are other effects that I am missing, please let me know.

I am also not happy with the list itself, as it is both very specific about certain parts while also being open-ended — but that is a debate we decided to postpone to a later PR.

@TimMoore The _Discouraged_ section has multiple bullet points and this is just the first of them — I assume that we do intend to make it clear that discrimination will not be tolerated? If so, then I don’t follow your argument. We also discussed earlier that the goal of this PR is to make the first part of the CoC purely positive to get people excited about the community; an anti-discrimination list does not contribute to this positivity.

Reply:
@rkuhn what I meant in my comment above is that the current wording is not an "anti-discrimination" list, it's an "explicitly welcome" list. It's primarily addressing the people who exist in various points of the multi-dimensional space it creates, particularly ones in areas that are frequently underrepresented in open source communities, or generally more subject to exclusion. Secondarily, it's reminding everyone to be sensitive to these differences.

This list was taken from the latest [Contributor Covenant](https://www.contributor-covenant.org/version/1/4/code-of-conduct). Here's how the Contributor Covenant FAQ [explains the list](https://www.contributor-covenant.org/faq#the-contributor-covenant-explicitly-lists-a-set-of-protected-classes-does-this-make-it-acceptable-to-discriminate-or-make-others-feel-unwelcome-based-on-other-factors):

> The Contributor Covenant explicitly lists a set of protected classes; does this make it acceptable to discriminate or make others feel unwelcome based on other factors?
>
> No. The Contributor Covenant explicitly lists protected classes for many reasons, such as reminding people to give them appropriate consideration, and assuring people in those protected classes that they are welcome. However, this is not an invitation for rules lawyers to seek loopholes, or to discriminate against others or make people feel unwelcome based on criteria not listed here. ([With the notable caveat that those who discriminate or make others feel unwelcome are themselves not welcome](https://en.wikipedia.org/wiki/Paradox_of_tolerance).)

To be honest, I consider it a small flaw in the Contributor Covenant that you need to also read the FAQ to fully understand the main document. I think we can improve on that here.

I'm open to the idea of reformulating the long list into something else that conveys the same intent in a different way. I'm not a huge fan of the list either: it will never be fully comprehensive, invites a lot of debate about what should or should not be listed, and distracts from the main point of the sentence. However, I think it is important to acknowledge the value of diversity and explicitly welcome people from underrepresented groups. That intent is what I don't want to lose.

If we can come up with a better wording for the introduction, then explicitly discouraging discrimination is also fine with me.